ubcpi/serialize.py

```diff
@@ -0,0 +1,248 @@
+IMAGE_ATTRIBUTES = {'position': 'image_position', 'show_image_fields': 'show_image_fields', 'alt': 'image_alt'}
+
+
+class UpdateFromXmlError(Exception):
+    """
+    Error occurred while deserializing the OpenAssessment XBlock content from XML.
+    """
+    pass
+
+
+class ValidationError(UpdateFromXmlError):
+    """
+    The XML definition is not semantically valid.
+    """
+    pass
+
+
+def _safe_get_text(element):
+    """
+    Retrieve the text from the element, safely handling empty elements.
+
+    Args:
+        element (lxml.etree.Element): The XML element.
+
+    Returns:
+        unicode
+    """
+    return unicode(element.text) if element.text is not None else u""
```

Comment:
Using a plain `unicode()` with no encoding will assume ASCII on Python 2.  This line of code will fail if element.text has any non-ASCII characters in it.  Even English can use non-ASCII if people use curly quotes or other fancy punctuation.